src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -11,44 +11,42 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class SymbolDisplayVisitor
     {
-        private void VisitTypeSymbolWithAnnotations(TypeSymbolWithAnnotations type, AbstractSymbolDisplayVisitor visitor = null)
+        private void VisitTypeSymbolWithAnnotations(TypeSymbolWithAnnotations type, AbstractSymbolDisplayVisitor visitorOpt = null)
         {
-            if (visitor == null)
+            var visitor = (SymbolDisplayVisitor)(visitorOpt ?? this.NotFirstVisitor);
+            var typeSymbol = type.TypeSymbol;
+            bool? isNullable = type.IsNullable;
+            if (typeSymbol.TypeKind == TypeKind.Array && isNullable.HasValue)
             {
-                visitor = this.NotFirstVisitor;
+                visitor.VisitArrayType((IArrayTypeSymbol)typeSymbol, isNullable: type.IsNullable);
             }
-
-            Debug.Assert(visitor is SymbolDisplayVisitor);
-
-            if (type.IsNullable == true)
+            else
             {
-                var typeSymbol = type.TypeSymbol;
-
-                if (typeSymbol.TypeKind == TypeKind.Array)
-                {
-                    ((SymbolDisplayVisitor)visitor).VisitArrayType((IArrayTypeSymbol)typeSymbol, isNullable: true);
-                }
-                else
+                typeSymbol.Accept(visitor);
+                switch (isNullable)
```

Comment:
Should this `switch` statement be extracted to a helper method and be re-used on line 101?

Reply:
There is a slight difference with the `switch` here adding an extra check for `typeSymbol.IsNullableType()`. Will leave as is for now.

---
In reply to: [171065571](https://github.com/dotnet/roslyn/pull/25087#discussion_r171065571) [](ancestors = 171065571)